app/src/main/java/xyz/whereuat/whereuat/MainActivity.java

```diff
@@ -179,23 +183,8 @@ public void onActivityResult(int req_code, int result_code, Intent data) {
 
         String name = con.getContactName();
         String phone = con.getPhoneNumber();
-        if (name != null && phone != null){
-            ContentValues values = new ContentValues();
-            values.put(ContactEntry.COLUMN_NAME, con.getContactName());
-            values.put(ContactEntry.COLUMN_PHONE, con.getPhoneNumber());
-            values.put(ContactEntry.COLUMN_AUTOSHARE, false);
-
-            InsertCommand cmd = new InsertCommand(this, ContactEntry.TABLE_NAME, null, values);
-            new DbTask(new DbTask.AsyncResponse() {
-                @Override
-                public void processFinish(Object result) {
-                    if ((Long) result != -1) {
-                        Log.d(TAG, "Successfully inserted contact");
-                    } else {
-                        Log.d(TAG, "Some weird things happened when inserting contact into DB");
-                    }
-                }
-            }).execute(cmd);
+        if (name != null && phone != null) {
+            new Contact(name, phone, false).dbInsert(this);
```

Comment:
Are you _sure_ this works? I think you need `(new Contact(name, phone, false)).dbInsert(this);`